### PR TITLE
Return 409 errors when accessing jobs in the queue

### PR DESCRIPTION
**Pull request by dbrower**
It is strange for jobs to "disappear" while they are being processed,
only to reappear again when finished. Also

 - unify the code handling files in the HTTP handlers.
 - adding handler tests to increase test coverage.

### CODE REVIEW

**Comment by dbrower** [comment]
@msuhovec Would you like to review this?

File: batchs/http_handlers.go [hunk 1]
```diff
@@ -134,23 +131,18 @@ func (s *RESTServer) SubmitJobIdHandler(w http.ResponseWriter, r *http.Request,
 
 	switch dir {
 	case "data", "error", "success":
-		jobPath := path.Join(s.QueuePath.basepath, dir, id)
-
-		if _, err := os.Stat(jobPath); err == nil {
```
Comment (msuhovec) [inline batchs/http_handlers.go]: Why is the stat unnecessary?

Comment (dbrower) [inline batchs/http_handlers.go]: If the directory doesn't exist then the `QueuePath.move` will fail anyway.

Comment (msuhovec) [inline batchs/http_handlers.go]: OK

**Pull request merged**